### PR TITLE
chore: add missing methods to intelicense class

### PR DESCRIPTION
**Pull request by joaoandre-avaiga**
This PR adds the public methods defined on the `Config` class to the .pyi file.

Please do a double check to make sure that I haven't forgotten anything.

Cheers.

### CODE REVIEW

File: src/taipy/config/config.pyi [hunk 1]
```diff
@@ -8,8 +8,45 @@
 # Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 # an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 # specific language governing permissions and limitations under the License.
-
 class Config:
```
Comment (trgiangdo) [inline src/taipy/config/config.pyi:24]:
```suggestion

import json
from typing import Any, Callable, Dict, List, Union, Optional

from .checker.issue_collector import IssueCollector
from .common._classproperty import _Classproperty
from .common._config_blocker import _ConfigBlocker
from .common.scope import Scope
from .common.frequency import Frequency
from .global_app.global_app_config import GlobalAppConfig
from .section import Section
from .unique_section import UniqueSection

class Config:
```

**Review by trgiangdo — DISMISSED** [review]
LGTM. This should cover all public methods of Config.

My only suggestion is to import all necessary typing so that the type hint is enable when call the methods.

File: src/taipy/config/config.pyi [hunk 2]
```diff
@@ -593,4 +625,3 @@ class Config:
         Returns:
             `TaskConfig^`: The default task configuration.
         """
```
Comment (jrobinAV) [inline src/taipy/config/config.pyi:627]: Please add a newline at EOF.

**Review by jrobinAV — COMMENTED** [review]
I guess this will need to be updated with the work on the config APIs you are currently implementing. 
I propose waiting for the issue https://github.com/Avaiga/taipy-config/issues/19 
to be done and the code merged to merge this PR.

Does it make sense to you?

**Comment by joaoandre-avaiga** [comment]
> I guess this will need to be updated with the work on the config APIs you are currently implementing. I propose waiting for the issue #19 to be done and the code merged to merge this PR.
> 
> Does it make sense to you?

I think we should merge this and I will add the new api changes to the config.pyi on the other PR. I think its a better history to have.

**Comment by jrobinAV** [comment]
> > I guess this will need to be updated with the work on the config APIs you are currently implementing. I propose waiting for the issue #19 to be done and the code merged to merge this PR.
> > Does it make sense to you?
> 
> I think we should merge this and I will add the new api changes to the config.pyi on the other PR. I think its a better history to have.

OK sounds good.

**Review by jrobinAV — APPROVED** [review]

**Review by trgiangdo — APPROVED** [review]

**Pull request merged**